optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/DroolsConstraint.java

```diff
@@ -57,13 +60,52 @@ public DroolsConstraint(DroolsConstraintFactory<Solution_> constraintFactory,
         return positive ? constraintWeight : constraintWeight.negate();
     }
 
-    public Rule createRule(Global<? extends AbstractScoreHolder> scoreHolderGlobal) {
-        List<RuleItemBuilder<?>> ruleItemBuilderList = new ArrayList<>(fromStreamList.size());
-        for (DroolsFromUniConstraintStream<Solution_, Object> fromStream : fromStreamList) {
-            fromStream.createRuleItemBuilders(ruleItemBuilderList, scoreHolderGlobal);
+    private Set<DroolsAbstractConstraintStream<Solution_>> assembleAllStreams() {
+        final Set<DroolsAbstractConstraintStream<Solution_>> streams = new LinkedHashSet<>();
+        for (DroolsAbstractConstraintStream<Solution_> fromStream: fromStreamList) {
+            streams.addAll(assembleAllStreams(fromStream));
         }
-        return PatternDSL.rule(constraintPackage, constraintName)
-                .build(ruleItemBuilderList.toArray(new RuleItemBuilder<?>[0]));
+        return streams;
+    }
+
+    private Collection<DroolsAbstractConstraintStream<Solution_>> assembleAllStreams(
+            DroolsAbstractConstraintStream<Solution_> parent) {
+        final Set<DroolsAbstractConstraintStream<Solution_>> streams = new LinkedHashSet<>();
```

Comment:
Code consistency: OptaPlanner's code uses plural for arrays and collections, *List for lists and *Set for sets.
So "streams" -> "streamSet"